### PR TITLE
detect/transform: from_base64 transform

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #10651 

This PR provides the `from_base64` transform used to match on content decoded with base64 using the mode (default rfc4648).

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [6487](https://redmine.openinfosecfoundation.org/issues/6487)

Describe changes:
- New transform: `from_base64`
- Documentation
- Keyword parser in rust.

Updates:
- Use Rust defined decode mode `DetectBase64Mode` everywhere to avoid ambiguity
- Rebase

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=pr/1670
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

File: src/util-decode-mime.c [hunk 1]
```diff
@@ -3162,7 +3162,7 @@ static int MimeBase64DecodeTest01(void)
         return 0;
 
     ret = DecodeBase64(dst, strlen(msg) + 1, (const uint8_t *)base64msg, strlen(base64msg),
-            &consumed_bytes, &num_decoded, BASE64_MODE_RFC2045);
+            &consumed_bytes, &num_decoded, ModeRFC2045);
```
Comment (catenacyber) [inline src/util-decode-mime.c:3165]: @jasonish do you know how cbindgen can export this as `Base64ModeRFC2045` ?

Comment (jasonish) [reply src/util-decode-mime.c:3165]: > @jasonish do you know how cbindgen can export this as `Base64ModeRFC2045` ?

Something like this? https://github.com/OISF/suricata/pull/8417/commits/857f23c17e472e3ef715e4d03f2a5d50f1b92c07

Comment (catenacyber) [reply src/util-decode-mime.c:3165]: I love that :-)

Comment (jasonish) [reply src/util-decode-mime.c:3165]: But it has this drawback which is why I didn't pursue it further then: https://github.com/OISF/suricata/pull/8417#issuecomment-1399210803

But with the library being a focus, we really need to start prefixing everything in global name space, so it should be revisited. Either having `cbindgen` do it, or just giving things a prefix in Rust when its not very rust-like.

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 19755

**Comment by jlucovsky** [comment]
Continued in #10707

**Pull request closed without merging**